packages/manager/src/views/WebPropertyEnvPage/components/EditAccess/EditMemberAccess.tsx

```diff
@@ -0,0 +1,206 @@
+import { useAddPermission, useDeleteMember } from '@app/services/rbac';
+import { toPascalCase } from '@app/utils/toPascalConvert';
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionToggle,
+  Button,
+  Checkbox,
+  Split,
+  SplitItem
+} from '@patternfly/react-core';
+import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import React, { useState } from 'react';
+import toast from 'react-hot-toast';
+interface Props {
+  onClose: () => void;
+  editMemberName: string;
+  propertyIdentifier: string;
+  memberList: any;
+}
+type ColumnNames = {
+  NAME: string;
+  APIKEY_CREATION: string;
+  APIKEY_DELETION: string;
+  PERMISSION_CREATION: string;
+  PERMISSION_DELETION: string;
+  ENV_CREATION: string;
+  ENV_SYNC: string;
+  [key: string]: string;
+};
+type GroupItem1 = {
+  email: string;
+  name: string;
+  role: string;
+  PERMISSION_CREATION: boolean;
+  PERMISSION_DELETION: boolean;
+  ENV_SYNC: boolean;
+  ENV_CREATION: boolean;
+  APIKEY_DELETION: boolean;
+  APIKEY_CREATION: boolean;
+};
+export const EditMemberAccess = ({
+  onClose,
+  editMemberName,
+  propertyIdentifier,
+  memberList
+}: Props): JSX.Element => {
+  const [expanded, setExpanded] = useState('def-list-toggle2');
+  const [addAccess, setAddAccess] = useState<string[]>([]);
+  const [deleteAccess, setDeleteAccess] = useState<string[]>([]);
+  const useAddPermission1 = useAddPermission(propertyIdentifier);
+  const deleteMember = useDeleteMember(propertyIdentifier);
+  const data1 = memberList.data.filter((e: GroupItem1) => e.name === editMemberName);
```

Comment:
data1 of what 😅 ...can you change the name from generic to something specific

Reply:
`editMemberName` is a prop and `memberList` is a prop. 

Then how about the prop memberList already gets the filtered item rather than again filtering. As i see memberlist is not even used anymore

Reply:
I can see that group is an object but there is only one key called data. 

How about we keep it simply as it is like `const [group,setGroup] = useState(data1)`